pkg/storage/engine.go

```diff
@@ -309,8 +357,42 @@ type IterOptions struct {
 	// use such an iterator is to use it in concert with an iterator without
 	// timestamp hints, as done by MVCCIncrementalIterator.
 	MinTimestampHint, MaxTimestampHint hlc.Timestamp
+	// KeyTypes specifies the types of keys to surface: point and/or range keys.
+	// Use HasPointAndRange() to determine which key type is present at a given
+	// iterator position, and RangeBounds() and RangeKeys() to access range keys.
+	// Defaults to IterKeyTypePointsOnly. For more info, see RangeKeys().
+	//
+	// NB: range keys are only supported for use with MVCCIterators, but it is
+	// legal to enable them for EngineIterators in order to derive cloned
+	// MVCCIterators from them. Range key behavior for EngineIterators is
+	// undefined.
+	KeyTypes IterKeyType
+	// RangeKeyMaskingBelow specifies a timestamp below which range keys will mask
```

Comment:
This doesn't pass 2AM test for me.
Maybe expand with:
```
// RangeKey with timestamp >= RangeKeyMaskingBelow will mask all point values with timestamp > RangeKeyMaskingBelow within its key range.
```
Otherwise it wasn't obvious if threshold applies to point values or to range key.

Reply:
Even better, I added a diagram with examples.